docs/api_guide/plot_06_sklearn_pipeline_cv_demo.py

```diff
@@ -0,0 +1,473 @@
+
+# %%
+pipeline = Pipeline(
+    [
+        (
+            "transformerbasis",
+            nmo.basis.TransformerBasis(nmo.basis.RaisedCosineBasisLinear(6)),
+        ),
+        (
+            "glm",
+            nmo.glm.GLM(regularizer_strength=0.5, regularizer="Ridge"),
+        ),
+    ]
+)
+
+pipeline.fit(X, y)
+
+# %%
+# Note how NeMoS models are already scikit-learn compatible and can be used directly in the pipeline.
+#
+# Visualize the fit:
+
+# %%
+
+# Predict the rate.
+# Note that you need a 2D input even if x is a flat array.
+# We are using expand dim to add the extra-dimension
+x = np.expand_dims(np.sort(X.flatten()), axis=1)
+predicted_rate = pipeline.predict(x)
+
+# %%
+fig, ax = plt.subplots()
+
+ax.scatter(X.flatten(), y, alpha=0.2, label="generated spike counts")
+ax.set_xlabel("input")
+ax.set_ylabel("spike count")
+
+
+ax.plot(
+    x,
+    predicted_rate,
+    label="predicted rate",
+    color="tab:orange",
+)
+
+ax.legend()
+sns.despine(ax=ax)
+
+# %%
+# There is some room for improvement, so in the next section we'll use cross-validation to tune the parameters of our pipeline.
+
+# %%
+# ### Select the number of basis by cross-validation
+
+# %%
+# !!! warning
+#     Please keep in mind that while `GLM.score` supports different ways of evaluating goodness-of-fit through the `score_type` argument, `pipeline.score(X, y, score_type="...")` does not propagate this, and uses the default value of `log-likelihood`.
+#
+#     To evaluate a pipeline, please create a custom scorer (e.g. `pseudo_r2` below) and call `my_custom_scorer(pipeline, X, y)`.
+#
+# #### Define the parameter grid
+#
+# Let's define candidate values for the parameters of each step of the pipeline we want to cross-validate. In this case the number of basis functions in the transformation step and the ridge regularization's strength in the GLM fit:
+
+# %%
+param_grid = dict(
+    glm__regularizer_strength=(0.1, 0.01, 0.001, 1e-6),
+    transformerbasis__n_basis_funcs=(3, 5, 10, 20, 100),
+)
+
+# %%
+# !!! note "Grid definition"
+#     In order to define a parameter grid dictionary for a pipeline, you must structure the dictionary keys as follows:
+#
+#     - Start with the pipeline label (`"glm"` or `"transformerbasis"` for us). This determines which pipeline step has the relevant hyperparameter.
+#     - Add `"__"` followed by the hyperparameter name (for example, `"n_basis_funcs"`).
+#     - If the hyperparameter is itself an object with attributes, add another `"__"` followed by the attribute name. For instance, `"glm__observation_model__inverse_link_function"`
+#       would be a valid key for cross-validating over the link function of the GLM's `observation_model` attribute `inverse_link_function`.
+#     The values in the dictionary are the parameters to be tested.
+
+
+
+# %%
+# #### Run the grid search
+# Let's run a 5-fold cross-validation of the hyperparameters with the scikit-learn `model_selection.GridsearchCV` class.
```

Comment:
should describe or link to a description of what "5-fold cross-validation" means